engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryConfigurationHelper.java

```diff
@@ -318,6 +320,32 @@ static PropertyNodeNameProvider determinePropertyNodeNameProvider(AbstractConfig
 		return new DefaultPropertyNodeNameProvider();
 	}
 
+	static LocaleResolver determineLocaleResolver(AbstractConfigurationImpl<?> hibernateSpecificConfig, Map<String, String> properties,
+			ClassLoader externalClassLoader) {
+		if ( hibernateSpecificConfig.getLocaleResolver() != null ) {
+			LOG.usingLocaleResolver( hibernateSpecificConfig.getLocaleResolver().getClass() );
+
+			return hibernateSpecificConfig.getLocaleResolver();
+		}
+
+		String localeResolverFqcn = properties.get( HibernateValidatorConfiguration.LOCALE_RESOLVER_CLASSNAME );
+		if ( localeResolverFqcn != null ) {
+			try {
+				@SuppressWarnings("unchecked")
+				Class<? extends LocaleResolver> clazz = (Class<? extends LocaleResolver>) run( LoadClass.action( localeResolverFqcn, externalClassLoader ) );
+				LocaleResolver localeResolver = run( NewInstance.action( clazz, "locale resolver class" ) );
+				LOG.usingLocaleResolver( clazz );
+
+				return localeResolver;
+			}
+			catch (Exception e) {
+				throw LOG.getUnableToInstantiatePropertyNodeNameProviderClassException( localeResolverFqcn, e );
```

Comment:
I think you need a dedicated logging method here; this one was copy-pasted.

Reply:
It was already there, I forgot to change the code.